test/apps/iodemo/io_demo.cc

```diff
@@ -480,13 +505,28 @@ class P2pDemoCommon : public UcxContext {
         uint16_t recalc_chksum = iov.calc_chksum();
         if (iov.chksum() != recalc_chksum) {
             pass = false;
-            LOG << "detected data corruption " << iov.sn() << " != "
+            LOG << "detected data corruption " << iov.chksum() << " != "
                 << recalc_chksum;
         }
 
         return pass;
     }
 
+    bool validate(const iomsg_t *msg) {
+        BufferIov *iov = _data_buffers_pool.get();
+
+        iov->init(opts().iomsg_size, _data_chunks_pool, 0, false);
+        iov->set_data(msg, opts().iomsg_size);
+
+        bool pass = validate(msg->tr.sn, *iov);
```

Comment:
need to use common func for validate to avoid data copy to buffer
(move to another PR, this PR will not validate iomsg payload)